telegram/bot.py

```diff
@@ -3457,7 +3457,8 @@ def answer_shipping_query(  # pylint: disable=C0103
         data: JSONDict = {'shipping_query_id': shipping_query_id, 'ok': ok}
 
         if ok:
-            assert shipping_options
+            if not shipping_options:
+                raise AssertionError
```

Comment:
What is the benefit of this except for maybe using `assert` in non-test files being considered bad practice? It basically does the same thing, but uses more words.

Reply:
Python has an option to compile the optimized bytecode and create the respective .pyo files by using the options `-O` and `-OO`.

During this, these basic optimizations are done:
- All the assert statements are removed
- All docstrings are removed (when `-OO` is selected)
- Value of the debug built-in variable is set to False

Now, if someone never intends to run the optimized bytecode, they can ignore this issue. But, it is still recommended not to have it in non-test files.
A better way for internal self-checks is to explicitly check and raise respective Error using an if statement.

Reply:
Fair enough. Please leave a short comment for that, though, along the lines "We don't user assert directly, as that's dropped in optimized bytecode". As a side note, please try to not force-push as that makes it hard to keep track of the PR ;)